pageserver/src/tenant.rs

```diff
@@ -786,6 +841,69 @@ impl Tenant {
         Ok(())
     }
 
+    /// Check for any local timeline directories that are temporary, or do not correspond to a
+    /// timeline that still exists: this can happen if we crashed during a deletion/creation, or
+    /// if a timeline was deleted while the tenant was attached to a different pageserver.
+    fn clean_up_timelines(&self, existent_timelines: &HashSet<TimelineId>) -> anyhow::Result<()> {
```

Comment:
More of an observation that should be reflected in the PR description & PR label: this is a breaking change.
Once we deploy code with this PR, there is no way to roll back because we don't write the attaching marker anymore => old code will treat incomplete attachment (:= interrupted load_remote_timelines) as completely attached and may start GC'ing data because it doesn't know about some timelines.

Reply:
Will pick this up in the other comment about forward compat below.